evaluation/evaluate.py

```diff
@@ -103,54 +104,20 @@ def evaluate_row(
     return output
 
 
-def run_evaluation(
-    openai_config: AzureOpenAIModelConfiguration,
-    testdata_path: Path,
-    results_dir: Path,
-    target_url: str,
-    passing_rate: int,
-    max_workers: int,
-    target_parameters: dict,
-    requested_metrics: list,
-    num_questions: int = None,
-):
-    """Run evaluation on the provided test data."""
-    logger.info("Running evaluation using data from %s", testdata_path)
-    testdata = load_jsonl(testdata_path)
-    if num_questions:
-        logger.info("Limiting evaluation to %s questions", num_questions)
-        testdata = testdata[:num_questions]
-
-    logger.info("Starting evaluation...")
-    for metric in requested_metrics:
-        if metric not in metrics_by_name:
-            logger.error(f"Requested metric {metric} is not available. Available metrics: {metrics_by_name.keys()}")
-            return False
-
-    requested_metrics = [
-        metrics_by_name[metric_name] for metric_name in requested_metrics if metric_name in metrics_by_name
-    ]
-
-    questions_with_ratings = []
-    with ThreadPoolExecutor(max_workers=max_workers) as executor:
-        futures = {
-            executor.submit(evaluate_row, row, target_url, openai_config, requested_metrics, target_parameters): row
-            for row in testdata
-        }
-        for future in track(concurrent.futures.as_completed(futures), description="Processing..."):
-            row_result = future.result()
-            questions_with_ratings.append(row_result)
-
-    logger.info("Evaluation calls have completed. Calculating overall metrics now...")
-    results_dir.mkdir(parents=True, exist_ok=True)
-
-    with open(results_dir / "eval_results.jsonl", "w", encoding="utf-8") as results_file:
-        for row in questions_with_ratings:
-            results_file.write(json.dumps(row, ensure_ascii=False) + "\n")
+def get_models(target_url: str) -> list:
+    """send request to /getmodels to determine whether the chosen model names are valid"""
+    try:
+        r = requests.get(target_url)
+        r.encoding = "utf-8"
+        r.raise_for_status()
 
-    dump_summary(questions_with_ratings, requested_metrics, passing_rate, results_dir)
-    plot_diagrams(questions_with_ratings, requested_metrics, passing_rate, results_dir)
-    return True
+        try:
+            response_list = r.json()
+        except json.JSONDecodeError:
+            raise ValueError(f"Response is not valid JSON:\n\n{r.text} \n")
```

Comment:
Formatting here seems excessive:
```suggestion
            raise ValueError(f"Response is not valid JSON:\n{r.text}")
```

Reply:
Why was this ignored and resolved without discussion :rofl:
The issue persists, it was just moved to `service_setup`